### PR TITLE
Migrate memory_android_chart.dart to null safety.

### PR DESCRIPTION
**Pull request by polina-c**

### CODE REVIEW

File: packages/devtools_app/lib/src/screens/memory/memory_android_chart.dart [hunk 1]
```diff
@@ -17,7 +15,8 @@ import 'memory_controller.dart';
 import 'memory_timeline.dart';
 
 class AndroidChartController extends ChartController {
-  AndroidChartController(this._memoryController, {List<int> sharedLabels})
+  AndroidChartController(this._memoryController,
+      {List<int> sharedLabels = const []})
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/screens/memory/memory_android_chart.dart:19]: nit: trailing comma, and add type to []

File: packages/devtools_app/lib/src/screens/memory/memory_android_chart.dart [hunk 2]
```diff
@@ -142,29 +141,30 @@ class MemoryAndroidChartState extends State<MemoryAndroidChart>
   void didChangeDependencies() {
     super.didChangeDependencies();
 
-    _memoryController = Provider.of<MemoryController>(context);
-
-    colorScheme = Theme.of(context).colorScheme;
+    final newMemoryController = Provider.of<MemoryController>(context);
+    if (_initialized && _memoryController == newMemoryController) return;
+    _memoryController = newMemoryController;
+    _initialized = false;
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/screens/memory/memory_android_chart.dart:147]: initialized = true

**Review by kenzieschmoll — APPROVED** [review]
lgtm with two comments

File: packages/devtools_app/lib/src/screens/memory/memory_android_chart.dart (hunk 1 above)
Comment (polina-c) [reply packages/devtools_app/lib/src/screens/memory/memory_android_chart.dart:19]: done

File: packages/devtools_app/lib/src/screens/memory/memory_android_chart.dart (hunk 2 above)
Comment (polina-c) [reply packages/devtools_app/lib/src/screens/memory/memory_android_chart.dart]: done

**Comment by polina-c** [comment]
tested

**Pull request merged**